### PR TITLE
new assemblies and modules for cluster

### PR DESCRIPTION
**Pull request by mburke5678**

### CODE REVIEW

**Comment by vikram-redhat** [comment]
@mburke5678 put the enterprise-4.0 label on. Let me know if not correct.

**Comment by mburke5678** [comment]
Preview:
http://file.rdu.redhat.com/~mburke/clusters-to-4.0/nodes/nodes-cluster-resource-levels.html

**Comment by mburke5678** [comment]
@derekwaynecarr 
I was asked to work on topics that would fit into a bucket called "Node." I went through the docs and identified topics that I thought were features that affected nodes or applied to nodes, etc.
Most of the text is unchanged, save for some changes to introductions and such.
I am looking for folks to take a higher-level look at what I did and let me know if these changes make sense for 4.0?
Are there any notable changes in 4.0 that affect these docs?
Do the user stories make sense?
Can you help me find someone(s) to help review this PR? I greatly appreciate any help.

**Comment by mburke5678** [comment]
Preview:
http://file.rdu.redhat.com/~mburke/clusters-to-4.0/nodes/nodes-cluster-resource-levels.html

File: modules/nodes-cluster-overcommit-configure-masters.adoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+object with default limits, per individual project or in the
+project template, in order to ensure that the overrides apply.
+
+
+.Procedure
+
+To configure a master for overcommit:
+
+. Configuring the `*ClusterResourceOverride*` admission controller in the
+*_master-config.yaml_* as in the following example (reuse the existing configuration tree
```
Comment (sjenning) [inline modules/nodes-cluster-overcommit-configure-masters.adoc]: nope

File: modules/nodes-cluster-overcommit-configure-masters.adoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
+// Module included in the following assemblies:
+//
+// * nodes/nodes-cluster-overcommit.adoc
+
+[id='nodes-cluster-overcommit-configure-masters_{context}']
+= Configuring masters for overcommitment
```
Comment (sjenning) [inline modules/nodes-cluster-overcommit-configure-masters.adoc]: need rewrite, if even possible, in 4.0

File: modules/nodes-cluster-overcommit-node-memory.adoc [hunk 3]
```diff
@@ -0,0 +1,43 @@
+// Module included in the following assemblies:
+//
+// * nodes/nodes-cluster-overcommit.adoc
+
+[id='nodes-cluster-overcommit-node-memory_{context}']
+
+= Reserving memory across quality of service tiers
+
+You can use the `experimental-qos-reserved` parameter to specify a percentage of memory to be reserved
```
Comment (sjenning) [inline modules/nodes-cluster-overcommit-node-memory.adoc]: this became not `experimental` upstream at some point.  not sure which release.

File: modules/nodes-cluster-overcommit-node-memory.adoc [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+by a pod in a particular QoS level. This feature attempts to reserve requested resources to exclude pods
+from lower OoS classes from using resources requested by pods in higher QoS classes.
+
+By reserving resources for higher QOS levels, pods that don't have resource limits are prevented from encroaching on the resources
+requested by pods at higher QoS levels.
+
+.Procedure
+
+To configure the `experimental-qos-reserved` parameter, edit the appropriate node configuration map.
```
Comment (sjenning) [inline modules/nodes-cluster-overcommit-node-memory.adoc]: node config map doesn't exist in 4.0

File: modules/nodes-cluster-overcommit-node-resources.adoc [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+.Procedure
+
+To explicitly reserve resources for non-pod processes, there are two
+ways to do so:
+
+- The preferred method is to allocate node resources by specifying resources
+available for scheduling. See Allocating Node Resources for more details.
+
+- Alternatively, you can create a *resource-reserver* pod that does nothing but
```
Comment (sjenning) [inline modules/nodes-cluster-overcommit-node-resources.adoc]: we should not document this.  this is a bad idea.  we should use `kube|system-reserved` via the kubelet.

File: modules/nodes-cluster-overcommit-qos-about.adoc [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+|3 (lowest)
+|*BestEffort*
+|If requests and limits are not set for any of the resources, then the container
+is classified as *BestEffort*.
+|===
+
+Memory is an incompressible resource, so in low memory situations, containers
+that have the lowest priority are terminated first:
+
+- *Guaranteed* containers are considered top priority, and are guaranteed to
```
Comment (sjenning) [inline modules/nodes-cluster-overcommit-qos-about.adoc]: we need to expand on this at some point.  there are significant downsides to `Guaranteed` not limited to unneeded OOMKilling and savage latency (a cpu: 10m pod can experience 90ms of scheduling latency) due to CPU throttling.

File: modules/nodes-cluster-overcommit-node-memory.adoc (hunk 3 above)
Comment (mburke5678) [reply modules/nodes-cluster-overcommit-node-memory.adoc]: @sjenning Is this the change you mean? Is the upstream change 3.12+ ?

File: modules/nodes-cluster-overcommit-configure-masters.adoc (hunk 2 above)
Comment (mburke5678) [reply modules/nodes-cluster-overcommit-configure-masters.adoc]: @sjenning 
>>need rewrite, if even possible, in 4.0

Are you suggesting that overcommit is not in 4.0?
Who would know for certain?

File: modules/nodes-cluster-overcommit-node-memory.adoc (hunk 4 above)
Comment (mburke5678) [reply modules/nodes-cluster-overcommit-node-memory.adoc]: @sjenning I am going to merge the docs as is to get these PRs in and will circle back and update for the 4.0 CRD functionality.

File: modules/nodes-cluster-overcommit-qos-about.adoc (hunk 6 above)
Comment (mburke5678) [reply modules/nodes-cluster-overcommit-qos-about.adoc]: @sjenning @smarterclayton I will merge this PR and make changes for 4.0 in a separate PR, unless anyone objects.

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL for modular issues. The specific wording in these topics might change or be removed, so a line-by-line review is not necessarily required.

**Comment by kalexand-rh** [comment]
@mburke5678, if you're happy with the modules, go head and rebase and merge. I think we need to include deeper peer reviews as part of the e2e review phase, but it's better to get the content in now.

File: modules/nodes-cluster-overcommit-configure-masters.adoc (hunk 2 above)
Comment (sjenning) [reply modules/nodes-cluster-overcommit-configure-masters.adoc]: let's not document this for now

File: modules/nodes-cluster-overcommit-qos-about.adoc (hunk 6 above)
Comment (mburke5678) [reply modules/nodes-cluster-overcommit-qos-about.adoc:45]: @sjenning @smarterclayton Any thoughts on what we want to document here? 

> there are significant downsides to Guaranteed

**Pull request merged**